docs/templates.md

```diff
@@ -0,0 +1,175 @@
+# Templates
+
+Sydent uses parametrised templates to generate the content of emails it
+sends and webpages it shows to users. Example templates can be found in
+the [res](https://github.com/matrix-org/sydent/tree/main/res) folder.
+
+## Branding
+
+Sending a value for `brand` to some API endpoints allows for different
+email and http templates to be used. These templates should be stored
+in a file structure like this:
+
+```
+ root_template_dir/
+     brand1/
+         [ brand1 template files ]
+     brand2/
+         [ brand2 template files ]
+```
+
+The config option `templates.root_directory` should be set to the path
+of `root_template_dir` and `templates.default_brand` should be set to
+the sub directory to use if no `brand` (or an invalid `brand`) is provided
+by the request.
+
+Here are the requests that can contain a value for `brand` and what
+template the brand value is used to select:
+
+Invite email templates:
+ - `POST /_matrix/identity/v2/validate/msisdn/requestToken`
+
+Verification email templates:
+ - `POST /_matrix/identity/v2/validate/email/requestToken`
+
+Verification SMS templates - BRAND CURRENTLY IGNORED:
+ - `POST /_matrix/identity/v2/validate/msisdn/requestToken`
+
+Verification response templates:
+ - `GET /_matrix/identity/v2/validate/email/submitToken`
+ - `GET /_matrix/identity/v2/validate/msisdn/submitToken`
+
+
+## Template formats
+
+### `.eml` template files
+
+Files ending in ".eml" can use python `%(variable)s` string substitution.
+
+Appending "_forurl" or "_forhtml" to any of the variable names listed below
+returns their values encoded suitably for URLs or HTML respectively.
+
+e.g. ">" in the HTML encoded version would be replaced with "&gt".
+
+### `.eml.j2` files
```

Comment:
Per some other conversation, we might want to include that these need to be `\r\n` line-endings?

Reply:
Is that in both the jinja and the .eml files?

Reply:
> these need to be \r\n line-endings?

oh god really. We ought to fix that in Sydent itself if we can...

Reply:
I don't think sydent should be expecting raw eml files, but that's a much larger issue.

Reply:
> Per some other conversation, we might want to include that these need to be `\r\n` line-endings?

That shouldn't be an issue... The libary Sydent uses says that "lone \r and \n characters are converted to \r\n characters" https://github.com/python/cpython/blob/1afc7b3219b24c951bb4e6b7e1ead904228de074/Lib/smtplib.py#L808

Reply:
This was fixed locally for the copies in ansible, I have not verified that they need to be `\r\n`, but there are comments referring to that.

Reply:
(This might not be true 🤷 )